native/src/main/java/io/ballerina/stdlib/mysql/Utils.java

```diff
@@ -32,32 +31,26 @@
  */
 public class Utils {
 
-    public static BMap generateOptionsMap(BMap mysqlOptions) {
-        if (mysqlOptions != null) {
-            BMap<BString, Object> options = ValueCreator.createMapValue();
-            addSSLOptions(mysqlOptions.getMapValue(Constants.Options.SSL), options);
+    public static void processOptionsMap(BMap mysqlOptions, BMap<BString, Object> options) {
+        addSSLOptions(mysqlOptions.getMapValue(Constants.Options.SSL), options);
 
-            long connectTimeout = getTimeout(mysqlOptions.get(Constants.Options.CONNECT_TIMEOUT));
-            if (connectTimeout > 0) {
-                options.put(Constants.DatabaseProps.CONNECT_TIMEOUT, connectTimeout);
-            }
-
-            long socketTimeout = getTimeout(mysqlOptions.get(Constants.Options.SOCKET_TIMEOUT));
-            if (socketTimeout > 0) {
-                options.put(Constants.DatabaseProps.SOCKET_TIMEOUT, socketTimeout);
-            }
-            
-            BString serverTimezone = mysqlOptions.getStringValue(Constants.Options.SERVER_TIMEZONE);
-            if (serverTimezone != null) {
-                options.put(Constants.DatabaseProps.SERVER_TIMEZONE, serverTimezone);
-            }
+        long connectTimeout = getTimeout(mysqlOptions.get(Constants.Options.CONNECT_TIMEOUT));
+        if (connectTimeout > 0) {
```

Comment:
Don't we need to  send a warning or error message when this value is less than 0?

Reply:
If it is less than 0, the driver will not timeout.